### PR TITLE
tela-icon-theme: remove darwin from platforms

### PR DESCRIPTION
**Pull request by figsoda**
###### Motivation for this change

ZHF: #144627

hydra failure: https://hydra.nixos.org/build/157846529/nixlog/2

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/icons/tela-icon-theme/default.nix [hunk 1]
```diff
@@ -36,7 +36,8 @@ stdenvNoCC.mkDerivation rec {
     description = "A flat colorful Design icon theme";
     homepage = "https://github.com/vinceliuice/tela-icon-theme";
     license = licenses.gpl3Only;
-    platforms = platforms.unix;
+    # darwin systems use case-insensitive filesystems that cause hash mismatches
+    platforms = subtractLists platforms.darwin platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/data/icons/tela-icon-theme/default.nix:40]: I think it would have been easier to set platforms to linux only.

**Comment by SuperSandro2000** [comment]
Also did you consider to remove/change the files that cause the hash mismatch?

**Comment by figsoda** [comment]
> Also did you consider to remove/change the files that cause the hash mismatch?

Yes, but it is pretty much impossible
```console
$ fd | sort | uniq -iD | wc -l
178
```

**Comment by SuperSandro2000** [comment]
Does this affect all 178 files? Oh well, whatever I guess.

File: pkgs/data/icons/tela-icon-theme/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/data/icons/tela-icon-theme/default.nix:40]: I would prefer to have other unix like systems to still be able to use it

**Comment by figsoda** [comment]
At least half of the 178 files needs to be removed to keep the hash mismatch and removing these files would also affect non-darwin usage

**Pull request merged**